src/implementation/Client/HTTPClient/HTTPClient.ts

```diff
@@ -43,6 +43,11 @@ export default class HTTPClient implements IClient {
     return CommunicationProtocolEnum.HTTP;
   }
 
+  async executeWithApiVersion(apiVersion: string = "v1.0", url: string, params: any = {}): Promise<object | string> {
```

Comment:
separate PR?

Reply:
Personally, rather than extracting this I would prefer to merge it now and fix it afterwards in a new PR. Bit difficult to get it out now :) thoughts?

Reply:
I think it is doable.

Reply:
Removed it